Foundation/NSURLCredential.swift

```diff
@@ -84,6 +105,20 @@ open class URLCredential : NSObject, NSSecureCoding, NSCopying {
         return self 
     }
     
+    open override func isEqual(_ object: Any?) -> Bool {
+        guard let other = object as? URLCredential else {
+            return false
+        }
+        
+        guard other !== self else {
```

Comment:
A style note, but this feels more like a traditional if/else than a guard to me. I'd like to preserve the use of guard for preconditions and such.